src/main.rs

```diff
@@ -219,13 +219,15 @@ mod tests {
         false
     }
 
-    #[test]
     #[allow(clippy::unit_cmp)]
```

Comment:
Not from this PR, but is `#[allow(clippy::unit_cmp)]` really required here?

Reply:
Yes, I had the same question. We need it for the final assert where we have a condition that it's always true. That is a workaround for printing messages to stdout.

Reply:
Fixed it.